manager/integration/tests/test_node.py

```diff
@@ -2239,4 +2256,282 @@ def test_node_eviction():
+        if replica.name == volume1_replica1.name or \
+           replica.name == volume1_replica2.name:
+            assert replica.running is True
+            assert replica.mode == "RW"
+        else:
+            volume1_err_replica = replica
+            break
+    assert volume1_err_replica is not None
+    assert volume1_err_replica.running is False
+    assert volume1_err_replica.mode == ''
+
+    client.update(node1, allowScheduling=False, evictionRequested=False)
+    wait_for_volume_replica_count(client, volume1_name, 2)
+
+    volume1 = client.by_id_volume(volume1_name)
+
+    for replica in volume1.replicas:
+        if replica.name == volume1_replica1.name or \
+           replica.name == volume1_replica2.name:
+            assert replica.running is True
+            assert replica.mode == "RW"
+
+    client.update(node1, evictionRequested=True)
+
+    replica_node_soft_anti_affinity_setting = \
+        client.by_id_setting(SETTING_REPLICA_NODE_SOFT_ANTI_AFFINITY)
+    try:
+        client.update(replica_node_soft_anti_affinity_setting,
+                      value="true")
+    except Exception as e:
+        print("\nException when update "
+              "Replica Node Level Soft Anti-Affinity setting",
+              replica_node_soft_anti_affinity_setting)
+        print(e)
+
+    wait_for_volume_replica_count(client, volume1_name, 3)
+    wait_for_volume_replica_count(client, volume1_name, 2)
+
+    volume1 = client.by_id_volume(volume1_name)
+    for replica in volume1.replicas:
+        assert replica.hostId == node2.name
+        assert replica.running is True
+        assert replica.mode == "RW"
+
+    client.update(node3, allowScheduling=True, evictionRequested=False)
+    client.update(node1, allowScheduling=True)
+
+    replica_node_soft_anti_affinity_setting = \
+        client.by_id_setting(SETTING_REPLICA_NODE_SOFT_ANTI_AFFINITY)
+    try:
+        client.update(replica_node_soft_anti_affinity_setting,
+                      value="false")
+    except Exception as e:
+        print("\nException when update "
+              "Replica Node Level Soft Anti-Affinity setting",
+              replica_node_soft_anti_affinity_setting)
+        print(e)
+
+    client.update(node2, allowScheduling=False, evictionRequested=True)
+
+    volume1 = client.by_id_volume(volume1_name)
+    assert volume1.robustness == "healthy"
+
+    wait_for_volume_replica_count(client, volume1_name, 3)
+    wait_for_volume_replica_count(client, volume1_name, 2)
+
+    volume1 = client.by_id_volume(volume1_name)
+    assert volume1.robustness == "healthy"
+
+    wait_for_volume_replica_count(client, volume1_name, 3)
+    wait_for_volume_replica_count(client, volume1_name, 2)
+
+    volume1 = client.by_id_volume(volume1_name)
+    assert volume1.robustness == "healthy"
+
+    for replica in volume1.replicas:
+        assert replica.hostId != node2.name
+        assert replica.running is True
+        assert replica.mode == "RW"
```

Comment:
Can we also check one replica on node-1, one on node-3? Thanks.

Reply:
fixed